### PR TITLE
Get mocha tests using mocha + fake reporter

### PR DESCRIPTION
**Pull request by mattfysh**

### CODE REVIEW

**Comment by Maciek416** [comment]
Nice! I know this has been a really really long time coming, so I hugely appreciate your patience and work on this.

The glue code this PR depends on has been merged in as magellan `6.1.0`: https://github.com/TestArmada/magellan/pull/101

**Comment by Maciek416** [comment]
Note: I added a test + CI stuff, along with a "mock suite" to test get_tests, filtering, etc. See:

https://github.com/TestArmada/magellan-mocha-plugin/pull/4

**Comment by Maciek416** [comment]
Hey @mattfysh , would you mind pushing another commit (a superficial commit is fine) to get this to run against the new test "suite" ? :)

**Comment by mattfysh** [comment]
will work on adding tests for this (along with https://github.com/TestArmada/magellan/pull/102) today :)

**Comment by mattfysh** [comment]
@Maciek416 have added test coverage for settings, iterator and filters (I think the groups filter may have been broken in master). I'll spend the next ~1hr adding tests for the TestRun class and then this should be ready, feel free to add comments on the existing diff so far. thanks :)

**Comment by Maciek416** [comment]
Nice! I'll take a look in detail shortly

**Comment by mattfysh** [comment]
@Maciek416 all done, let me know once you've had a chance to pass over the code. We also wrote a bamboo reporter as part of this case, does magellan support reporter plugins? happy to publish `magellan-bamboo` to the registry

**Comment by Maciek416** [comment]
@mattfysh hey Matt, I've just whipped up this project to explain how a reporter is structured, what the events mean, how to figure out if a test is starting for the first time, retrying, or failing for the final time. It also shows how to get the stdout/stderr of individually-executing test runs, and how to install your reporter (whether as a local file or public module):

https://github.com/TestArmada/magellan-reporter-example-outline

File: test_support/coffee/spec.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+describe 'Suite', ->
```
Comment (Maciek416) [inline test_support/coffee/spec.coffee:1]: Awesome. Thanks for this test

File: lib/get_tests.js [hunk 2]
```diff
@@ -1,56 +1,35 @@
-var _ = require("lodash");
-var acorn = require("acorn");
-var walk = require("acorn/dist/walk");
+var fs = require('fs');
 var path = require("path");
-var fs = require("fs");
+var spawnSync = require('spawn-sync');
```
Comment (Maciek416) [inline lib/get_tests.js]: Does our linter not catch the single quotes here?

File: lib/get_tests.js [hunk 3]
```diff
@@ -1,56 +1,35 @@
...
 var Locator = require("./locator");
-
 var mochaSettings = require("./settings");
+var reporter = path.resolve(__dirname, 'test_capture.js');
```
Comment (Maciek416) [inline lib/get_tests.js]: (ditto for quotes here and elsewhere)

**Comment by Maciek416** [comment]
This is looking great. I had a comment about linting rules but I realized we don't lint this project yet, and I don't want to bog down this PR with code cleanup. We can just do that in a subsequent sweep.

File: lib/get_tests.js [hunk 4]
```diff
@@ -1,56 +1,35 @@
...
-  }).map(function (f) {
-    return f.trim();
-  });
+  if (mochaSettings.mochaOpts) {
+    args.push('--opts', mochaSettings.mochaOpts);
+  }
+
+  args = args.concat(mochaSettings.mochaTestFolders);
+
+  process.env.MOCHA_CAPTURE_PATH = OUTPUT_PATH;
```
Comment (Maciek416) [inline lib/get_tests.js]: We should merge into the new env object instead of writing to `process.env`

**Comment by Maciek416** [comment]
Ok, I think I just have the one last comment about writing to `process.env` and then she'll be ready to rock.

**Comment by mattfysh** [comment]
@Maciek416 awesome - nice catch! ready for re-review.

**Comment by mattfysh** [comment]
re: linting -- yeah I noticed my style was off from yours but that we didn't have any linting so I thought a follow-up case to mend consistency is probably the best approach.

**Comment by Maciek416** [comment]
Published as `6.0.0`

**Pull request merged**